mvcc/backend/backend.go

```diff
@@ -536,3 +557,16 @@ func (s *snapshot) Close() error {
 	<-s.donec
 	return s.Tx.Rollback()
 }
+
+func copyTxReadBuffer(from txReadBuffer) txReadBuffer {
+	to := txReadBuffer{txBuffer: txBuffer{buckets: make(map[string]*bucketBuffer)}}
+	for k, v := range from.txBuffer.buckets {
```

Comment:
copy looks correct to me. Extract the bucket copy logic into a `func (bucketBuffer) copy() *bucketBuffer` function?

Reply:
sounds good.